pkg/resolver/search.go

```diff
@@ -21,9 +21,17 @@ import (
 	"github.com/stolostron/search-v2-api/pkg/config"
 	db "github.com/stolostron/search-v2-api/pkg/database"
 	"github.com/stolostron/search-v2-api/pkg/metric"
+	"github.com/stolostron/search-v2-api/pkg/rbac"
 	"k8s.io/klog/v2"
 )
 
+// Stuct to keep a copy of users access
+type UserResourceAccess struct {
+	CsResources     []rbac.Resource            // Cluster-scoped resources on hub the user has list access.
+	NsResources     map[string][]rbac.Resource // Namespaced resources on hub the user has list access.
+	ManagedClusters []string                   // Managed clusters where the user has view access.
+}
```

Comment:
Should move this definition to `userData.go` ?  
It can help to stay in sync with the `UserData` struct.

Reply:
Actually, we should use inheritance (or composition?) to represent these 2 structs better. I haven't used it before, so we don't have an example, but it should be something like this...

```go
type UserAccessData struct {
	CsResources     []rbac.Resource            // Cluster-scoped resources on hub the user has list access.
	NsResources     map[string][]rbac.Resource // Namespaced resources on hub the user has list access.
	ManagedClusters []string                   // Managed clusters where the user has view access.
}

type UserDataCache struct {
	UserAccessData

	// Add all the fields to manage the cache.
	// ...
}
```

Reply:
moved